spring-core/src/main/java/org/springframework/core/codec/CharSequenceEncoder.java

```diff
@@ -75,12 +79,32 @@ public boolean canEncode(ResolvableType elementType, @Nullable MimeType mimeType
 					return Hints.getLogPrefix(hints) + "Writing " + formatted;
 				});
 			}
-			CharBuffer charBuffer = CharBuffer.wrap(charSequence);
-			ByteBuffer byteBuffer = charset.encode(charBuffer);
-			return bufferFactory.wrap(byteBuffer);
+			boolean release = true;
+			int capacity = calculateCapacity(charSequence, charset);
+			DataBuffer dataBuffer = bufferFactory.allocateBuffer(capacity);
+			try {
+				dataBuffer.write(charSequence, charset);
+				release = false;
+			}
+			catch (IllegalStateException ex) {
```

Comment:
Shouldn't this be changed to `CoderMalfunctionError`, if I remember correctly?

Reply:
Ah right, missed that part. I'll force push the branch.